### ISSUE TITLE
MessageConsumerPact matchingRules empty

### ISSUE THREAD

**Issue by madmaxrichter** [pact-foundation/pact-js#635]
### Software versions

- **OS**: Linux Mint 20.1 (Kernel 5.4.0-67-generic)
- **Consumer Pact library**: 9.15.3
- **Provider Pact library**: pact-jvm-provider-spring_2.12 v 3.6.1
- **Node Version**: v12.21.0

### Issue Checklist

Please confirm the following:

- [x] I have upgraded to the latest
- [x] I have the read the FAQs in the Readme
- [x] I have triple checked, that there are **no unhandled promises** in my code
- [x] I have set my log level to debug and attached a log file showing the complete request/response cycle
- [ ] For bonus points and virtual high fives, I have created a reproduceable git repository (see below) to illustrate the problem

### Expected behaviour

the `matchingRules` property of generated contract .json file should contain some rules according to the defined values and types

```json
{
  "consumer": {
    "name": "customer-authentication"
  },
  "provider": {
    "name": "customers-service-messaging"
  },
  "messages": [
    {
      "description": "a customer-updated message",
      "providerStates": [
        {
          "name": "customer messaging",
          "params": null
        }
      ],
      "contents": {
        "changes": [
          {
            "fieldName": "email",
            "oldValue": "old@email.com",
            "newValue": "new@email.com"
          }
        ],
        "customer": {
          "email": "new@email.com"
        }
      },
      "matchingRules": {
        "body": {
          "$.changes": {
            "matchers": [
              {
                "min": 1
              }
            ]
          },
          "$.changes[*].*": {
            "matchers": [
              {
                "match": "type"
              }
            ]
          },
          "$.customer.email": {
            "matchers": [
              {
                "match": "type"
              }
            ]
          }
        }
      },
      "metaData": {
        "content-type": "application/json"
      }
    }
  ],
  "metadata": {
    "pactSpecification": {
      "version": "3.0.0"
    }
  }
}
```

### Actual behaviour

`matchingRules`  is an empty object

```json
{
  "consumer": {
    "name": "customer-authentication"
  },
  "provider": {
    "name": "customers-service-messaging"
  },
  "messages": [
    {
      "description": "a customer-updated message",
      "providerStates": [
        {
          "name": "customer messaging",
          "params": null
        }
      ],
      "contents": {
        "changes": {
          "contents": {
            "fieldName": {
              "contents": "email",
              "json_class": "Pact::SomethingLike"
            },
            "oldValue": {
              "contents": "old@email.com",
              "json_class": "Pact::SomethingLike"
            },
            "newValue": {
              "contents": "new@email.com",
              "json_class": "Pact::SomethingLike"
            }
          },
          "json_class": "Pact::ArrayLike",
          "min": 1
        },
        "customer": {
          "email": {
            "contents": "new@email.com",
            "json_class": "Pact::SomethingLike"
          }
        }
      },
      "matchingRules": {
      },
      "metaData": {
        "content-type": "application/json"
      }
    }
  ],
  "metadata": {
    "pactSpecification": {
      "version": "3.0.0"
    }
  }
}
```

### Steps to reproduce

I can reproduce this for multiple repositories that our company is using pact in. It only affects messageConsumers written in node/js. It starts breaking for us when upgrading from pact-js 9.1.1 to any newer version. What I am suspecting is the fact that this is related to underlying pact-node version which changed after 9.1.1. from 9 to 10.
The sample code on my machine though works fine, so I assume it could be some issue due to combination of other node modules used in all our projects. I am happy to privide any specific debug output that helps to pin down the root cause of this.

### Relevant log files

[pact_messaging_consumer.log](https://github.com/pact-foundation/pact-js/files/6189402/pact_messaging_consumer.log)

**Comment by anto-ac** [comment]
We've just come across the same issue! Downgrading to 9.1.1 worked!

**Comment by mefellows** [comment]
Thanks folks, apologies for the delay on this - we'll get it looked into.

**Comment by madmaxrichter** [comment]
> Thanks folks, apologies for the delay on this - we'll get it looked into.

Awesome, let me know if I could help or provide further input

**Comment by bethesque** [comment]
I think it's related to https://github.com/pact-foundation/pact-ruby-standalone/issues/65

Sorry, I haven't had an OSS day for weeks 😱 I'll see if I can bump it up the priority list.

**Comment by mefellows** [comment]
It does look like that. The thing that stands out is that the Ruby JSON format is in there, instead of the expanded matchers. Glad you know what it is Beth!

**Comment by bethesque** [comment]
Ok, that turned out to be less scary than I thought. I'd accepted a PR that unintentionally changed the existing JSON.load to JSON.parse, so the Ruby matching objects weren't being deserialised into their correct classes. The updated standalone should come out of the Ruby Goldberg release machine any minute.

**Comment by TimothyJones** [comment]
Thanks @bethesque! I'll release pact-node today (because of some reasons, it's a little more fiddly than it's supposed to be at the moment).

**Comment by TimothyJones** [comment]
`@pact-foundation/pact-node@10.12.2` is now out with this fix. If you remove package-lock and do another npm install, you should get it with pact-js.

**Comment by mefellows** [comment]
Thanks Tim!

**Comment by TimothyJones** [comment]
Now releasing `@pact-foundation/pact` version `9.15.5` which pins the version of `pact-node` to get the fix. It should be out in 10 mins or so.

I'll close this for now, please feel free to reopen if the issue isn't resolved in 9.15.5.

**Comment by madmaxrichter** [comment]
> Now releasing `@pact-foundation/pact` version `9.15.5` which pins the version of `pact-node` to get the fix. It should be out in 10 mins or so.
> 
> I'll close this for now, please feel free to reopen if the issue isn't resolved in 9.15.5.

Thanks, I can confirm the issue is fixed for us now :)
